compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -479,17 +491,35 @@ impl<'a> Parser<'a> {
         self.token.token() == &Token::EOF
     }
 
-    fn span_since(&self, start_span: Span) -> Span {
-        if self.current_token_span == start_span {
+    fn location_since(&self, start_location: Location) -> Location {
+        // When taking the span between locations in different files, just keep the first one
+        if self.current_token_location.file != start_location.file {
+            return start_location;
+        }
+
+        let start_span = start_location.span;
+
+        let span = if self.current_token_location.span == start_location.span {
             start_span
         } else {
-            let end_span = self.previous_token_span;
-            Span::from(start_span.start()..end_span.end())
-        }
+            let end_span = self.previous_token_location.span;
+            if start_span.start() <= end_span.end() {
+                Span::from(start_span.start()..end_span.end())
+            } else {
+                // TODO: workaround for now
```

Comment:
Please add a follow-up issue for this TODO

Reply:
Captured in https://github.com/noir-lang/noir/issues/7453